roles/ceph-osd/tasks/standard.yml

```diff
@@ -4,15 +4,15 @@
   shell: ceph osd crush add-bucket {{ ssd_pool }} root
   register: result
   changed_when: result.stdout |search("added bucket")
-  when: inventory_hostname == groups['ceph_osds_ssd'][0]
+  when: inventory_hostname == (play_hosts | intersect(groups['ceph_osds_ssd']))[0]
```

Comment:
First, I think no role will include ceph-osd role as a dependency.
But let's assume that there is a role named ceph-osd-child which depends on ceph-osd role.
I am not sure if you mean following examples:
1. All in one play
```
- name: all in one play
  hosts: ceph_osds
  roles:
    - role: ceph-osd
    - role: ceph-osd-child
```
Yes, if there is a play like this, the PR can't prevent is running twice. But I think no one will write a play like this. As ceph-osd is a dependency of ceph-osd-child, no need to run ceph-osd individually. Running ceph-osd-child is good enough here.

2. In two plays
```
- name: play one
  hosts: ceph_osds
  roles:
    - role: ceph-osd

- name: play two
  hosts: ceph_osds
  roles:
    - role: ceph-osd-child
```
In this case, the task will running twice too. But to my knowledge, running twice also happen with `run_once` in this case. Because the two roles are in two plays.

Following change is for running with limit, e.g, `-l ceph2,ceph3`, no ceph1(`groups['ceph_osds_ssd'][0]`)
```
-  when: inventory_hostname == groups['ceph_osds_ssd'][0]
+  when: inventory_hostname == (play_hosts | intersect(groups['ceph_osds_ssd']))[0]
```